src/uptane/secondary.cc

```diff
@@ -59,18 +58,24 @@ Json::Value Secondary::newTargetsCallBack(const std::vector<Uptane::Target> &tar
 }
 
 void Secondary::setKeys(const std::string &public_key, const std::string &private_key) {
-  if (!boost::filesystem::exists(config.full_client_dir)) {
-    boost::filesystem::create_directories(config.full_client_dir);
+  boost::filesystem::path private_parent_dir = boost::filesystem::path(config.ecu_private_key()).parent_path();
+  if (!private_parent_dir.empty()) {
+    boost::filesystem::create_directories(private_parent_dir);
```

Comment:
We might want to think about adding an optional enum second parameter to Utils::writeFile to optionally create any necessary parent directories.

Reply:
I am in favor of that!